state-chain/runtime/src/chainflip/bitcoin_elections.rs

```diff
@@ -337,6 +374,18 @@ impl Hooks<BitcoinBlockHeightTracking, BitcoinDepositChannelWitnessing> for Bitc
 			>,
 		>(deposit_channel_witnessing_identifiers.clone(), &chain_progress)?;
 
+		// I think we should always use this as our upper limit to avoid not reaching consensus in
+		// case there is a reorg, using this block means safety margin will be kept into account for
+		// this election
```

Comment:
```suggestion
		// We use `ProcessedUpTo` as our upper limit to avoid not reaching consensus in
		// case there is a reorg, using this block means safety margin will be kept into account for
		// this election, and thus are much less likely to ask nodes to query for a block they don't have.
```